### DISCUSSION TITLE
New imagemagick command option -highwater (or -extreme)

### DISCUSSION THREAD

**Discussion by reader3** [ImageMagick/ImageMagick#4361] [Digital Image Processing]
After working several hours with layering, flattening, compositing, I found out that what I am actually looking for is something like a highwater mark for each pixel in an image file. In other words the maximum (or minimum) RGB values for an individual pixel location is kept, until all images are processed, and whatever maximum (minimum) RGB value occurred is fixed for that pixel.

All this came about because I was trying to catch a white lightning flash on a black background, but unfortunately when flattening or layer images, when I used 1000, they all produced a pure white picture which gave a strong hint to me of what was going on. I tried inverted colors too, but everything kept coming up white.

Has anyone come across this problem before, of trying to capture the maximum intensity and color value for a pixel locatoin and saving it (or minimum value)

Anyways:
convert -highwater image1 image2 image3 ... final.image
is what I had in mind
or perhaps to catch the minimum values too
convert -extreme minimum image1 image2 image3 ... final.image

**Comment by fmw42** [comment]
Not sure I understand what you want. Is this what you want  http://www.fmwconcepts.com/imagemagick/maxrgb/index.php

**Reply by fmw42** [reply]
You can also do

`convert image1 image2 image3 -evaluate-sequence max result`

**Comment by snibgo** [comment]
What version of IM, on what platform?

> ... but unfortunately when flattening or layer images, when I used 1000, ...

What commands did you use?

As fmw42 says. Or it can be done with:
```
magick image1.png image2.png image3.png -background Black -channel RGB,sync -compose Lighten -layers Flatten output.png
```
If you have many large images that won't fit into your memory at the same time, you can split the command like this:
```
magick image1.png image2.png -evaluate-sequence max image3.png -evaluate-sequence max output.png
```
